### PR TITLE
Generate json schema

### PR DESCRIPTION
**Pull request by simitt**
## Motivation/summary
This PR removes the manually crafted JSON schemas for the Intake API. 

Previous refactoring of the Intake decoder made JSON schemas obsolete in the decoder process. They are still kept around as they are an integral part of the APM agent development cycle, where automated tests are run against the schemas. To ensure the JSON schemas do not get out of sync with the actual decoder logic, this PR introduces a code generator that builds the JSON schemas, based on the internally used go structs, their validation tags and comments. 

Whenever new fields are added running `make update` will re-generate the JSON schemas and keep them in sync. 

## Checklist

~- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).~
- [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

I have considered changes for:
- [x] documentation
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
- [x] automated tests (add tests for the code changes, all [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally)
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes
Add/remove/update fields, comments and implemented validation tags and run `make update` and verify the changes are accordingly reflected in the JSON schemas. 

## Related issues
#4334

### CODE REVIEW

**Review by axw — COMMENTED** [review]
Basically LGTM - main question I have is about the TODO. The others are nits or opinions that can be dismissed at will.

File: model/modeldecoder/generator/interface.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+				if err := generateJSONPropertyInteger(info, parent, child); err != nil {
+					return err
+				}
+			case TypeNameNumber:
+				if err := generateJSONPropertyJSONNumber(info, parent, child); err != nil {
+					return err
+				}
+			case TypeNameObject:
+				child.Type.add(TypeNameObject)
+			//TODO(implement):
```
[the remark this answers is not in the document]
Comment (simitt) [reply model/modeldecoder/generator/interface.go:55]: actually that was an old TODO from before implementing the cases - removed it now.

File: model/modeldecoder/generator/generatortest/model.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	Nullable nullable.Bool `json:"nullable"`
+}
+
+type HTTPHeader struct {
+	Required nullable.HTTPHeader `json:"required" validate:"required"`
+	Nullable nullable.HTTPHeader `json:"nullable"`
+}
+
+type Interface struct {
+	Required nullable.Interface `json:"required" validate:"required,inputTypes=string;int;float64;bool;map[string]interface,maxLength=5,minLength=2,pattern=patternA,max=250,min=1.5"`
```
[the remark this answers is not in the document]
Comment (simitt) [reply model/modeldecoder/generator/generatortest/model.go:56]: yes that will be more aligned

File: model/modeldecoder/generator/parser.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	return pkgs[0], nil
+}
+
+func parse(pkg *packages.Package, parsed *Parsed) error {
+	for _, file := range pkg.Syntax {
+		// parse type comments
+		typeComments := make(map[int]string)
+		for _, c := range file.Comments {
+			typeComments[int(c.End())] = trimComment(c.Text())
+		}
```
[the remark this answers is not in the document]
Comment (simitt) [reply model/modeldecoder/generator/parser.go:96]: I did, but I don't see how I could use it here to simplify the comment mapping.

**Review by axw — APPROVED** [review]

**Comment by simitt** [comment]
@axw I removed the generated go json schemas and adapted the package tests that still used them. Also found one bug where the `maxLength` was not set (will backport this fix to >= 7.10) once the tests were update. Can you please review the last commits.

**Comment by axw** [comment]
Even better! LGTM.

**Pull request merged**